jsapp/js/components/map.es6

```diff
@@ -150,15 +156,21 @@ export class FormMap extends React.Component {
 
     if (viewby) {
       var mapMarkers = this.prepFilteredMarkers(this.state.submissions, this.props.viewby);
-      var mM = [];
-      let choices = this.props.asset.content.choices;
+      var mM = [], lbl = undefined;
+      let choices = this.props.asset.content.choices,
+          survey = this.props.asset.content.survey;
+
+      let q = survey.find(z => z.name === viewby || z.$autoname === viewby);
 
       Object.keys(mapMarkers).map(function(m, i) {
-        var lbl = choices.find(o => o.name === m || o.$autoname == m);
+        if (q && q.type == 'select_one') {
+          lbl = choices.find(o => o.list_name === q.select_from_list_name && (o.name === m || o.$autoname === m));
+        }
         mM.push({
           count: mapMarkers[m].count,
           id: mapMarkers[m].id,
-          labels: lbl ? lbl.label : undefined
+          labels: lbl ? lbl.label : undefined,
+          value: m != "undefined" ? m : undefined
```

Comment:
Does it mean that `m` can actually be a string "undefined"? Wouldn't that cause errors in few previous lines? Maybe we should return function on top if `m` has bad value?

Reply:
No, it wouldn't cause errors, but it would print "undefined" to users when rendering the list. But the "undefined" string can still be a property in the previous lines (this comes from the API call).